### PR TITLE
bpo-39583: Remove superfluous "extern C" bits from Include/cpython/*.h

### PR DESCRIPTION
**Pull request by smontanaro**
See: https://mail.python.org/archives/list/python-dev@python.org/message/QP23KZZKISBIIP76AHPX7WFKMOOX2UCQ/

Normal build/test pass, e.g.,

./configure && make -j && make test

I tried configuring with CC=g++ but that generated all kinds of fireworks during build. I'm sure it's possible to build using a C++ compiler, but I'm no longer facile with that C dialect, so don't know all the buttons to press in configure.

https://bugs.python.org/issue39583

### CODE REVIEW

**Comment by smontanaro** [comment]
I don't understand the docs failure. This PR doesn't have any doc changes, so my guess is it's caused by something else.

File: Misc/NEWS.d/next/C API/2020-02-08-08-01-35.bpo-39583.qURKSl.rst
Comment (vstinner) [inline Misc/NEWS.d/next/C API/2020-02-08-08-01-35.bpo-39583.qURKSl.rst]: To not confused Sphinx on `*` character:

```suggestion
Remove superfluous "extern C" declarations from ``Include/cpython/*.h``.
```

**Review by vstinner — APPROVED** [review]
LGTM since these header files cannot be included directly. They are only included in other header files where extern "C" is already used.

**Comment by vstinner** [comment]
Ah, the Documentation job of the CI failed:

```
Warning, treated as error:
../build/NEWS:169:Inline emphasis start-string without end-string.
```

See my comment on your NEWS entry.

**Comment by vstinner** [comment]
> I don't understand the docs failure. This PR doesn't have any doc changes, so my guess is it's caused by something else.

NEWS entries are part of the documentation: https://docs.python.org/dev/whatsnew/changelog.html

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
Please resolve conflicts.

**Comment by csabella** [comment]
@smontanaro. it looks like you have approval, but this merge conflicts need to be resolved first.  Thanks!

**Comment by smontanaro** [comment]
I think I straightened most things out, but it needs a bit more tweaking to build properly. I'm afraid the underlying assumption that all these extern C bits weren't required might have gone by the boards.

**Comment by smontanaro** [comment]
@vstinner or @serhiy-storchaka (or some other committer) Feel free to merge (not a capability I have or want). Thanks...

**Comment by vstinner** [comment]
I fixed a merge conflict.

**Comment by vstinner** [comment]
Merged, thanks @smontanaro!

**Pull request merged**
